raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -276,6 +299,17 @@ private void onBecomeLeader(long currentTimeMs) {
         resetConnections();
 
         kafkaRaftMetrics.maybeUpdateElectionLatency(currentTimeMs);
+
+        accumulator = new BatchAccumulator<>(
```

Comment:
@hachikuji Should we move this state to `LeaderState`? We can delegate `maybeCloseAccumulator` to the `transitionTo...` functions.

I have a similar requirements for snapshot. I am currently adding a `SnapshotWriter` to `FollowerState` to track this state. I haven't implemented it yet but I am thinking of making `FollowerState` `Closeable` which will handle the cleanup. The `transitionTo...` functions have the additional responsibility of calling `close` if necessary.

What do you think?